### PR TITLE
Creating formal schema definitions for misc config files

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

File: schema/pubber_configuration.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      "default": "RS256"
+    },
+    "serialNo": {
+      "type": "string"
+    },
+    "macAddr": {
+      "type": "string"
+    },
+    "keyBytes": {
+      "description": "Should be type byte[], but that can't be represented in json schema"
```
Comment (johnrandolph) [inline schema/pubber_configuration.json:33]: What type is it then?

**Comment by grafnu** [comment]
The java code generator creates it as a generic Object, and then explicit
type-casts are used to make it byte[] as needed. I updated the comment to
add that bit of detail.

File: pubber/src/main/java/daq/pubber/Pubber.java [hunk 2]
```diff
@@ -261,6 +265,13 @@ private static void startFeedListener(String projectId, String siteName, String
     }
   }
 
+  private static PubberConfiguration sanitizeConfiguration(PubberConfiguration configuration) {
```
Comment (johnrandolph) [inline pubber/src/main/java/daq/pubber/Pubber.java:268]: this is more like build, right?

**Review by johnrandolph — APPROVED** [review]

**Comment by grafnu** [comment]
It's actually taking an existing object in cleaning it up (thus sanizing)
-- not just building one from scratch!

**Pull request merged**